app/services/create_newsletter_signup.rb

```diff
@@ -30,13 +28,15 @@ def sendgrid_api_call
     api_call_data = {
       'list_ids' => list_ids,
       'contacts' => [
-        'first_name' => first_name,
-        'last_name' => last_name,
-        'email' => email
+        'first_name' => newsletter_signup[0].newsletter_signup[:first_name],
+        'last_name' => newsletter_signup[0].newsletter_signup[:last_name],
+        'email' => newsletter_signup[0].newsletter_signup[:email]
       ]
```

Comment:
@ColinW520 I hate the code in this block. I tried to manipulate the array to get to something more elegant but haven't figured out anything better.